cluster/etcd_elector.go

```diff
@@ -0,0 +1,104 @@
+// +build kvdb_etcd
+
+package cluster
+
+import (
+	"context"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/concurrency"
+	"github.com/coreos/etcd/pkg/transport"
+	"github.com/lightningnetwork/lnd/channeldb/kvdb/etcd"
+)
+
+const (
+	// etcdConnectionTimeout is the timeout until successful connection to
+	// the etcd instance.
+	etcdConnectionTimeout = 10 * time.Second
+)
+
+// Enforce that etcdLeaderElector implements the LeaderElector interface.
+var _ LeaderElector = (*etcdLeaderElector)(nil)
+
+// etcdLeaderElector is an implemetation of LeaderElector using etcd as the
+// election gowernor.
+type etcdLeaderElector struct {
+	id       string
```

Comment:
Should the ID be unique for each node in the cluster, or is it mean to uniquely identify a pod?

Reply:
Yes the ID must be unique cluster wide.